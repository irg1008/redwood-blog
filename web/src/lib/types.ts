export type KeysEndingWith<T, Suffix extends string> = Extract<
  keyof T,
  `${string}${Suffix}`
>

export type KeysStartingWith<T, Prefix extends string> = Extract<
  keyof T,
  `${Prefix}${string}`
>
