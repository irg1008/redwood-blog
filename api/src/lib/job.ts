import { WorkerUtilsOptions, quickAddJob } from 'graphile-worker'

type JobParams = Parameters<typeof quickAddJob>
type Args = JobParams extends [WorkerUtilsOptions, ...infer U] ? U : never

const options: WorkerUtilsOptions = {
  connectionString:
    process.env.NODE_ENV === 'test'
      ? process.env.TEST_DATABASE_URL
      : process.env.DATABASE_URL,
}

export const job = (...args: Args) => quickAddJob(options, ...args)
