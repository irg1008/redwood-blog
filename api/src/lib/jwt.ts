import {
  JSONWebKeySet,
  JWK,
  JWTPayload,
  SignJWT,
  createLocalJWKSet,
  exportJWK,
  exportPKCS8,
  generateKeyPair,
  importPKCS8,
  jwtVerify,
} from 'jose'

export const ALG = 'PS256'
export const KID = 'base'

export const createKeyPair = async () => {
  const { privateKey, publicKey } = await generateKeyPair(ALG)
  return {
    privateKey: await exportPKCS8(privateKey),
    publicKey: await exportJWK(publicKey),
  }
}

export const privateKey = async () => {
  return await importPKCS8(process.env.JWT_PRIVATE_KEY, ALG)
}

export const signJwt = async (
  payload: JWTPayload,
  signCb?: (sign: SignJWT) => void
) => {
  const key = await privateKey()
  const sign = new SignJWT(payload)
  if (signCb) signCb(sign)
  return sign.setProtectedHeader({ alg: ALG, kid: KID }).setIssuedAt().sign(key)
}

export const verifyJwt = async (token: string) => {
  const key = await privateKey()
  return jwtVerify(token, key)
}

export const publicVerifyJwt = async (token: string) => {
  const jwks = createLocalJWKSet(mediaJWKS)
  return jwtVerify(token, jwks)
}

const mediaJWK: JWK = {
  kid: KID,
  kty: 'RSA',
  e: 'AQAB',
  n: process.env.JWT_PUBLIC_KEY,
  alg: ALG,
}

export const mediaJWKS: JSONWebKeySet = {
  keys: [mediaJWK],
}
