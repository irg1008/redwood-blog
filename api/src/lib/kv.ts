import Redis, { RedisOptions } from 'ioredis'

export const redisOptions: RedisOptions = {
  port: parseInt(process.env.REDIS_PORT),
  password: process.env.REDIS_HOST_PASSWORD,
  host: process.env.REDIS_HOST,
}

export const KV = new Redis(redisOptions)
