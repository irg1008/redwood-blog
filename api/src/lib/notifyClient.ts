import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client'
import { setContext } from '@apollo/client/link/context'
import { SendServerEventInput, ServerEventResult, User } from 'types/graphql'

import { signPayload } from '@redwoodjs/api/webhooks'

const link = createHttpLink({
  uri: `http://0.0.0.0:${process.env.API_PORT}/graphql`,
})

const authLink = setContext((req, { headers }) => {
  const workerSingnature = signPayload('timestampSchemeVerifier', {
    payload: JSON.stringify(req.variables),
    secret: process.env.WORKER_SECRET,
  })

  return {
    headers: {
      ...headers,
      [process.env.WORKER_SIGNATURE]: workerSingnature,
    },
  }
})

const apolloClient = new ApolloClient({
  link: authLink.concat(link),
  cache: new InMemoryCache(),
})

const SEND_SERVER_EVENT = gql`
  mutation SendServerEvent($input: SendServerEventInput!) {
    sendServerEvent(input: $input) {
      message
      topic
    }
  }
`

export const notifyClient = async (
  userId: User['id'],
  topic: string,
  data: unknown
) => {
  return apolloClient.mutate<
    ServerEventResult,
    { input: SendServerEventInput }
  >({
    mutation: SEND_SERVER_EVENT,
    variables: {
      input: {
        topic,
        userId,
        message: JSON.stringify(data),
      },
    },
  })
}
