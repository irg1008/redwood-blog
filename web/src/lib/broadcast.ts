import i18next from 'i18next'

export const listenI18nBroadcast = () => {
  const i18nBC = new BroadcastChannel('i18n')

  i18next.on('languageChanged', () => {
    i18nBC.postMessage(i18next.language)
  })

  i18nBC.onmessage = (event) => {
    const lang = event.data
    if (lang === i18next.language) return
    i18next.changeLanguage(event.data)
  }
}
