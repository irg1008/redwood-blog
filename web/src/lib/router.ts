export const deleteSearchParams = (...params: string[]) => {
  const url = new URL(window.location.href)
  params.forEach((param) => {
    url.searchParams.delete(param)
  })
  window.history.pushState({}, '', url)
}
