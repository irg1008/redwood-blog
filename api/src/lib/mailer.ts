import { Mailer } from '@redwoodjs/mailer-core'
import { NodemailerMailHandler } from '@redwoodjs/mailer-handler-nodemailer'
import { ReactEmailRenderer } from '@redwoodjs/mailer-renderer-react-email'

import { logger } from 'src/lib/logger'

export const DOMAIN = 'example.com'

export const mailDirections = {
  noReply: `no-reply@${DOMAIN}`,
  contactReceiver: `contact@${DOMAIN}`,
} satisfies Record<string, `${string}@${typeof DOMAIN}`>

export const mailer = new Mailer({
  handling: {
    handlers: {
      nodemailer: new NodemailerMailHandler({
        transport: {
          host: process.env.SMTP_HOST,
          port: process.env.SMTP_PORT,
          secure: process.env.SMTP_PORT === '465',
          auth: {
            user: process.env.SMTP_USER,
            pass: process.env.SMTP_PASS,
          },
        },
      }),
    },
    default: 'nodemailer',
  },

  rendering: {
    renderers: {
      reactEmail: new ReactEmailRenderer(),
    },
    default: 'reactEmail',
  },

  logger,

  defaults: {
    replyTo: mailDirections.noReply,
    from: `"No Reply" <${mailDirections.noReply}>`,
  },
})
