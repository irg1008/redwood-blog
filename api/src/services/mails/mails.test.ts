import { InMemoryMailHandler } from '@redwoodjs/mailer-handler-in-memory'

import { mailer } from 'src/lib/mailer'

import {
  sendConfirmUserEmail,
  sendContactEmail,
  sendResetPasswordEmail,
} from './mails'

describe('mails', () => {
  beforeEach(() => {
    const handler = mailer.getTestHandler() as InMemoryMailHandler
    handler.clearInbox()
  })

  it('sends reset password email', async () => {
    const resetData = {
      email: 'jane@doe.com',
      resetToken: '123456',
    }

    await sendResetPasswordEmail(resetData)

    const mailHandler = mailer.getTestHandler() as InMemoryMailHandler
    expect(mailHandler.inbox).toHaveLength(1)

    const [sentEmail] = mailHandler.inbox

    expect({
      ...sentEmail,
      htmlContent: undefined,
      textContent: undefined,
    }).toMatchInlineSnapshot(`
      {
        "attachments": [],
        "bcc": [],
        "cc": [],
        "from": ""No Reply" <no-reply@example.com>",
        "handler": "nodemailer",
        "handlerOptions": undefined,
        "headers": {},
        "htmlContent": undefined,
        "renderer": "reactEmail",
        "rendererOptions": {},
        "replyTo": "no-reply@example.com",
        "subject": "Reset your password",
        "textContent": undefined,
        "to": [
          "jane@doe.com",
        ],
      }
    `)

    expect(sentEmail.htmlContent).toMatchSnapshot()
    expect(sentEmail.textContent).toMatchSnapshot()

    const restLinkRegex = new RegExp(
      `/reset-password\\?resetToken=${resetData.resetToken}`,
      'g'
    )
    expect(sentEmail.htmlContent.match(restLinkRegex)).toHaveLength(2)
  })

  it('sends confirm account email', async () => {
    const confirmData = {
      email: 'jane@doe.com',
      code: 123456,
    }

    await sendConfirmUserEmail(confirmData)

    const mailHandler = mailer.getTestHandler() as InMemoryMailHandler
    expect(mailHandler.inbox).toHaveLength(1)

    const [sentEmail] = mailHandler.inbox

    expect({
      ...sentEmail,
      htmlContent: undefined,
      textContent: undefined,
    }).toMatchInlineSnapshot(`
      {
        "attachments": [],
        "bcc": [],
        "cc": [],
        "from": ""No Reply" <no-reply@example.com>",
        "handler": "nodemailer",
        "handlerOptions": undefined,
        "headers": {},
        "htmlContent": undefined,
        "renderer": "reactEmail",
        "rendererOptions": {},
        "replyTo": "no-reply@example.com",
        "subject": "Confirm your account",
        "textContent": undefined,
        "to": [
          "jane@doe.com",
        ],
      }
    `)

    expect(sentEmail.htmlContent).toMatchSnapshot()
    expect(sentEmail.textContent).toMatchSnapshot()

    expect(sentEmail.htmlContent).toContain(confirmData.code.toString())

    const codeRegex = new RegExp(`\n${confirmData.code}\n`, 'g')
    expect(sentEmail.textContent.match(codeRegex)).not.toBeNull()
  })

  it('sends contact email', async () => {
    const contactData = {
      email: 'jane@doe.com',
      message: 'Hello!',
      name: 'Jane Doe',
    }

    await sendContactEmail(contactData)

    const mailHandler = mailer.getTestHandler() as InMemoryMailHandler
    expect(mailHandler.inbox).toHaveLength(1)

    const [sentEmail] = mailHandler.inbox

    expect({
      ...sentEmail,
      htmlContent: undefined,
      textContent: undefined,
    }).toMatchInlineSnapshot(`
      {
        "attachments": [],
        "bcc": [],
        "cc": [],
        "from": ""No Reply" <no-reply@example.com>",
        "handler": "nodemailer",
        "handlerOptions": undefined,
        "headers": {},
        "htmlContent": undefined,
        "renderer": "reactEmail",
        "rendererOptions": {},
        "replyTo": "jane@doe.com",
        "subject": "New Contact Form Submission",
        "textContent": undefined,
        "to": [
          "contact@example.com",
        ],
      }
    `)

    expect(sentEmail.htmlContent).toMatchSnapshot()
    expect(sentEmail.textContent).toMatchSnapshot()
  })
})
