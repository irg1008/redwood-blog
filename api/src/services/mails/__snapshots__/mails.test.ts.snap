// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mails sends confirm account email 2`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html dir="ltr" lang="en">

  <head>
    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />
    <meta name="x-apple-disable-message-reformatting" />
  </head>
  <div style="display:none;overflow:hidden;line-height:1px;opacity:0;max-height:0;max-width:0">Confirm action<div> ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿</div>
  </div>

  <body style="margin-left:auto;margin-right:auto;margin-top:auto;margin-bottom:auto;background-color:rgb(255,255,255);font-family:ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;">
    <table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em;margin-left:auto;margin-right:auto;margin-top:40px;margin-bottom:40px;border-radius:0.25rem;border-width:1px;border-style:solid;border-color:rgb(229,231,235);padding:20px">
      <tbody>
        <tr style="width:100%">
          <td>
            <h1 style="margin-left:0px;margin-right:0px;margin-top:30px;margin-bottom:30px;padding:0px;text-align:center;font-size:24px;font-weight:400;color:rgb(0,0,0)">Please use this code to confirm your action</h1>
            <hr style="width:100%;border:none;border-top:1px solid #eaeaea;margin-left:0px;margin-right:0px;margin-top:26px;margin-bottom:26px;border-width:1px;border-style:solid;border-color:rgb(234,234,234)" />
            <p style="font-size:14px;line-height:24px;margin:16px 0;color:rgb(0,0,0)">123456</p>
            <p style="font-size:14px;line-height:24px;margin:16px 0;color:rgb(0,0,0)">If you didn&#x27;t request this code, you can ignore this email.</p>
          </td>
        </tr>
      </tbody>
    </table>
  </body>

</html>"
`;

exports[`mails sends confirm account email 3`] = `
"PLEASE USE THIS CODE TO CONFIRM YOUR ACTION

--------------------------------------------------------------------------------

123456

If you didn't request this code, you can ignore this email."
`;

exports[`mails sends contact email 2`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html dir="ltr" lang="en">

  <head>
    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />
    <meta name="x-apple-disable-message-reformatting" />
  </head>
  <div style="display:none;overflow:hidden;line-height:1px;opacity:0;max-height:0;max-width:0">New contact form submitted<div> ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿</div>
  </div>

  <body style="margin-left:auto;margin-right:auto;margin-top:auto;margin-bottom:auto;background-color:rgb(255,255,255);font-family:ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;">
    <table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em;margin-left:auto;margin-right:auto;margin-top:40px;margin-bottom:40px;border-radius:0.25rem;border-width:1px;border-style:solid;border-color:rgb(229,231,235);padding:20px">
      <tbody>
        <tr style="width:100%">
          <td>
            <h1 style="margin-left:0px;margin-right:0px;margin-top:30px;margin-bottom:30px;padding:0px;text-align:center;font-size:24px;font-weight:400;color:rgb(0,0,0)">New contact from Jane Doe &lt;jane@doe.com&gt;</h1>
            <hr style="width:100%;border:none;border-top:1px solid #eaeaea;margin-left:0px;margin-right:0px;margin-top:26px;margin-bottom:26px;border-width:1px;border-style:solid;border-color:rgb(234,234,234)" />
            <p style="font-size:14px;line-height:24px;margin:16px 0;color:rgb(0,0,0)">Hello!</p>
          </td>
        </tr>
      </tbody>
    </table>
  </body>

</html>"
`;

exports[`mails sends contact email 3`] = `
"NEW CONTACT FROM JANE DOE <JANE@DOE.COM>

--------------------------------------------------------------------------------

Hello!"
`;

exports[`mails sends reset password email 2`] = `
"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html dir="ltr" lang="en">

  <head>
    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type" />
    <meta name="x-apple-disable-message-reformatting" />
  </head>
  <div style="display:none;overflow:hidden;line-height:1px;opacity:0;max-height:0;max-width:0">Reset your password<div> ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿</div>
  </div>

  <body style="margin-left:auto;margin-right:auto;margin-top:auto;margin-bottom:auto;background-color:rgb(255,255,255);font-family:ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Roboto, &quot;Helvetica Neue&quot;, Arial, &quot;Noto Sans&quot;, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Symbol&quot;, &quot;Noto Color Emoji&quot;">
    <table align="center" width="100%" border="0" cellPadding="0" cellSpacing="0" role="presentation" style="max-width:37.5em;margin-left:auto;margin-right:auto;margin-top:40px;margin-bottom:40px;border-radius:0.25rem;border-width:1px;border-style:solid;border-color:rgb(229,231,235);padding:20px">
      <tbody>
        <tr style="width:100%">
          <td>
            <h1 style="margin-left:0px;margin-right:0px;margin-top:30px;margin-bottom:30px;padding:0px;text-align:center;font-size:24px;font-weight:400;color:rgb(0,0,0)">Reset the password for jane@doe.com</h1>
            <hr style="width:100%;border:none;border-top:1px solid #eaeaea;margin-left:0px;margin-right:0px;margin-top:26px;margin-bottom:26px;border-width:1px;border-style:solid;border-color:rgb(234,234,234)" />
            <p style="font-size:14px;line-height:24px;margin:16px 0;color:rgb(0,0,0)">Click on the link below to reset your password.</p><a href="http://localhost:8910/reset-password?resetToken=123456" style="color:rgb(0,0,0);text-decoration:none;font-size:14px;line-height:24px" target="_blank">Reset password</a>
            <p style="font-size:14px;line-height:24px;margin:16px 0;color:rgb(0,0,0)">If you can&#x27;t click the link, copy and paste the following URL into
              your browser:</p>
            <p style="font-size:14px;line-height:24px;margin:16px 0;word-break:break-all;color:rgb(0,0,0)">http://localhost:8910/reset-password?resetToken=123456</p>
          </td>
        </tr>
      </tbody>
    </table>
  </body>

</html>"
`;

exports[`mails sends reset password email 3`] = `
"RESET THE PASSWORD FOR JANE@DOE.COM

--------------------------------------------------------------------------------

Click on the link below to reset your password.

Reset password http://localhost:8910/reset-password?resetToken=123456

If you can't click the link, copy and paste the following URL into your browser:

http://localhost:8910/reset-password?resetToken=123456"
`;
